python/tests/tsutil.py

```diff
@@ -539,6 +539,85 @@ def compute_mutation_parent(ts):
     return mutation_parent
 
 
+def py_subset(tables, nodes, record_provenance=True):
+    """
+    Naive implementation of the TableCollection.subset method using the Python API.
+    """
+    if np.any(nodes > tables.nodes.num_rows) or np.any(nodes < 0):
+        raise ValueError("Nodes out of bounds.")
+    full = tables.copy()
+    # there is no table collection clear in the py API
+    tables.nodes.clear()
+    tables.individuals.clear()
+    tables.populations.clear()
+    tables.edges.clear()
+    tables.migrations.clear()
+    tables.sites.clear()
+    tables.mutations.clear()
+    # mapping from old to new ids
+    node_map = {}
+    ind_map = {tskit.NULL: tskit.NULL}
+    pop_map = {tskit.NULL: tskit.NULL}
+    for old_id in nodes:
+        node = full.nodes[old_id]
+        if node.individual not in ind_map and node.individual != tskit.NULL:
+            ind = full.individuals[node.individual]
+            new_ind_id = tables.individuals.add_row(
+                ind.flags, ind.location, ind.metadata
+            )
+            ind_map[node.individual] = new_ind_id
+        if node.population not in pop_map and node.population != tskit.NULL:
+            pop = full.populations[node.population]
+            new_pop_id = tables.populations.add_row(pop.metadata)
+            pop_map[node.population] = new_pop_id
+        new_id = tables.nodes.add_row(
+            node.flags,
+            node.time,
+            pop_map[node.population],
+            ind_map[node.individual],
+            node.metadata,
+        )
+        node_map[old_id] = new_id
+    for edge in full.edges:
+        if edge.child in nodes and edge.parent in nodes:
+            tables.edges.add_row(
+                edge.left,
+                edge.right,
+                node_map[edge.parent],
+                node_map[edge.child],
+                edge.metadata,
+            )
+    for mig in full.migrations:
+        if mig.node in nodes:
+            tables.migrations.add_row(
+                mig.left,
+                mig.right,
+                node_map[mig.node],
+                pop_map.get(mig.source, tskit.NULL),
```

Comment:
I don't think this works - see comments above in the C. I think we should just not bother with migrations.